pkg/nfd-master/nfd-master.go

```diff
@@ -346,6 +399,43 @@ func removeLabels(n *api.Node, labelNames []string) {
 	}
 }
 
+// getExtendedResourceOps returns a slice of operations to perform on the node status
+func getExtendedResourceOps(n *api.Node, extendedResources ExtendedResources) []statusOp {
+	var statusOps []statusOp
+
+	oldResources := strings.Split(n.Annotations[AnnotationNs+"extended-resources"], ",")
+
+	// figure out which resources to remove
+	for _, resource := range oldResources {
+		// provided that the old resource exists in the node
```

Comment:
I just realized that we need to do similar handling of the labels namespace/prefix that we do in `addLabels(). Worker can send labels either in non-namespaced format (in which case the namespace is added automatically), or, in the full namespaced format.